tests/ttnn/unit_tests/operations/test_line_all_gather_nightly.py

```diff
@@ -0,0 +1,263 @@
+        mem_config,
+        use_program_cache,
+        function_level_defaults,
+        enable_async,
+        num_iters,
+    )
+
+
+def run_line_all_gather_instances(
+    t3k_device_mesh,
+    num_devices,
+    num_instances,
+    input_shape,
+    dim,
+    num_links,
+    input_dtype,
+    layout,
+    mem_config,
+    use_program_cache,
+    function_level_defaults,
+    enable_async,
+    num_iters=1,
+):
+    if t3k_device_mesh.get_num_devices() != 8:
+        pytest.skip("Not T3000!")
+
+    for device in t3k_device_mesh.get_devices():
+        device.enable_async(enable_async)
+
+    logger.info(f"Input shape: {input_shape}")
+    logger.info(f"dim: {dim}")
+
+    (is_known_failure, message) = is_unsupported_case(
+        input_shape, dim, mem_config, num_devices, num_links, input_dtype, layout
+    )
+    if is_known_failure:
+        pytest.skip(f"Skipping unsupported case {message}.")
+
+    t3k_device = []
+
+    for device in t3k_device_mesh.get_devices():
+        t3k_device.append(device)
+
+    t3000_device_rows = [
+        [t3k_device[4], t3k_device[0], t3k_device[3], t3k_device[7]],
+        [t3k_device[5], t3k_device[1], t3k_device[2], t3k_device[6]],
+    ]
+    logger.info(f"Input shape: {input_shape}")
+    logger.info(f"dim: {dim}")
+
+    input_tensor = torch.rand(input_shape).bfloat16()
+
+    ttnn_tensor = ttnn.from_torch(input_tensor, mesh_mapper=ShardTensorToMesh(t3k_device_mesh, dim=dim))
+    input_tensor_mesh = ttnn.to_device(ttnn_tensor, t3k_device_mesh)
+
+    result_mesh_tensors = []
+    for i, devices in enumerate(t3000_device_rows):
+        tt_out_tensor = ttnn.line_all_gather(input_tensor_mesh, dim, num_links=num_links, memory_config=mem_config)
+        result_mesh_tensors.append(tt_out_tensor)
+
+    ## Wait for completion
+    for i, devices in enumerate(t3000_device_rows):
+        for d in devices:
+            ttl.device.Synchronize(d)
+
+    for tt_out_tensor in result_mesh_tensors:
+        for i, t in enumerate(ttnn.get_device_tensors(tt_out_tensor)):
+            tt_output_tensor = t.cpu().to(ttl.tensor.Layout.ROW_MAJOR).to_torch()
+            if input_dtype == ttl.tensor.DataType.BFLOAT16:
+                eq, output = comp_equal(tt_output_tensor, input_tensor)
+            else:
+                eq, output = comp_pcc(tt_output_tensor, input_tensor)
+            if not eq:
+                logger.error(f"output mismatch for tensor {i}")
+            assert eq, f"{i} FAILED: {output}"
+
+
+@skip_for_grayskull("Requires eth connected devices to run")
+@pytest.mark.parametrize(
+    "num_devices, num_instances, num_links, input_shape, dim, layout",
+    [
+        # (4, 1, [4, 1, 33, 256], 0, ttl.tensor.Layout.ROW_MAJOR),
+        # (8, 1, [8, 1, 33, 256], 0, ttl.tensor.Layout.ROW_MAJOR),
+        # # (8, 1, [8, 1, 256, 32], 0, ttl.tensor.Layout.TILE),
+        # (8, 1, [8, 8, 256, 384], 1, ttl.tensor.Layout.ROW_MAJOR),
+        # (4, 2, [8, 8, 256, 384], 1, ttl.tensor.Layout.TILE),
```

Comment:
Is there a reason these other 4-chip cases aren't enabled? The 8chip cases being disabled makes sense (and they should just be removed from the list), but all 4-chip cases here should be enabled.